File: 283-move-zeroes/283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int zeroes = 0, n = nums.length;
        for(int num : nums)
            if(num == 0) zeroes++;
        int nonZeroes = n - zeroes;
        
        int eIdx = 0;
        for(int i = 0; i < n; ++i) 
            if(nums[i] != 0) nums[eIdx++] = nums[i];
        while(eIdx < n) 
            nums[eIdx++] = 0;
    }
}

File: 283-move-zeroes/283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        zeroes, n = 0, len(nums)
        for num in nums:
            if num == 0: zeroes += 1
        nonZeroes, eIdx =  n - zeroes, 0
        for i in range(n): 
            if nums[i] != 0: 
                nums[eIdx] = nums[i]
                eIdx += 1
        while eIdx < n:
            nums[eIdx] = 0
            eIdx += 1
        